resolver/resolver_test.go

```diff
@@ -242,7 +252,26 @@ func TestHandler(t *testing.T) {
 	if m.Rcode != 3 {
 		t.Error("not setting NXDOMAIN for AAAA requests")
 	}
+}
+
+func TestHandlerWithDomainPatterns(t *testing.T) {
+	res, err := fakeDNSwithPatterns(8053, &[]string{"test-{{.tname}}"})
+	if err != nil {
+		t.Error(err)
+	}
 
+	dns.HandleFunc("mesos.", res.HandleMesos)
+	go res.Serve("udp")
+	go res.Serve("tcp")
+
+	// wait for startup ? lame
```

Comment:
Very good. Arbitrary sleeps to synchronize code hurt the eyes. What's the status of #146? Looks like your changes and mine will conflict, so we have to adapt this or #146, whichever is merged earlier.